doodle-formats/src/format/deflate.rs

```diff
@@ -30,6 +30,17 @@ fn bits_value_u8(name: &'static str, n: usize) -> Expr {
     }
 }
 
+fn dist_value_u8(name: &'static str, n: usize) -> Expr {
+    if n > 1 {
+        bit_or(
+            tuple_proj(var(name), n - 1),
+            shl_u8(dist_value_u8(name, n - 1), 1),
+        )
+    } else {
+        tuple_proj(var(name), 0)
+    }
+}
+
```

Comment:
Maybe we could rename this and the existing bits function to indicate their endianness explicitly?

Reply:
It's a bit complicated to specify the endianness, since the bits are written from LSB-to-MSB within each byte during the bitstream-to-bytestream conversion. The endianness of distance-codes is then 'natural-order' in the order a prefix code would normally be written, and proper 'numbers' have their bits written in order of LSB-to-MSB.

So one could argue that distance-codes are big-endian in a little-endian framing, or little-endian in a big-endian framing.
And equally as well, that numbers are little-endian in a little-endian frame, or big-endian in a big-endian frame.

So naming them closer to the 'bit-order' semantics like the dichotomies of  'code' vs 'number' or 'natural' vs 'reversed' would perhaps be clearer? Idk.

Reply:
I do agree clearer names would be useful, but the complexity of the problem-space makes it hard to come up with unambiguous labels for the two modes of bit-writing.